src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -812,11 +812,20 @@ protected void runMayThrow()
         return futures;
     }
 
+    /**
+     * Forces a major compaction of specified token ranges of the specified column family.
+     * <p>
+     * The token ranges will be interpreted as closed intervals to match the closed interval defined by the first and
+     * last keys of a sstable, even though the {@link Range} class is suppossed to be half-open by definition.
+     *
+     * @param cfStore The column family store to be compacted.
+     * @param ranges The token ranges to be compacted, interpreted as closed intervals.
+     */
     public void forceCompactionForTokenRange(ColumnFamilyStore cfStore, Collection<Range<Token>> ranges)
     {
         Callable<Collection<AbstractCompactionTask>> taskCreator = () -> {
             Collection<SSTableReader> sstables = sstablesInBounds(cfStore, ranges);
-            if (sstables == null || sstables.isEmpty())
+            if (sstables.isEmpty())
```

Comment:
This change is unrelated to the patch and my personal preference is to stick to the main patch in older versions

Reply:
Fine with me. The unneeded null check isn't on trunk nor 4.1, so it will remain on 3.11 and 4.0.